checker/src/main/java/org/checkerframework/checker/nullness/NullnessNoInitVisitor.java

```diff
@@ -844,7 +842,7 @@ public Void visitNewClass(NewClassTree tree, Void p) {
         if (enclosingExpr != null) {
             checkForNullability(enclosingExpr, DEREFERENCE_OF_NULLABLE);
         }
-        AnnotatedDeclaredType type = atypeFactory.getAnnotatedType(tree);
+        //        AnnotatedDeclaredType type = atypeFactory.getAnnotatedType(tree);
```

Comment:
Here, the old code again looks at the tree, to get the explicit annotations.
Like for arrays, can you change this to just look at the type? With improved handling in the ATF, looking at the tree should no longer be needed.